src/analyzer.rs

```diff
@@ -9,437 +9,503 @@ use std::collections::{BTreeMap, HashMap};
-            Some(format!("{}{}", stack, key.to_upper_camel_case()))
-        }
-        "" => {
-            if s.x_kubernetes_int_or_string.is_some() {
-                Some("IntOrString".into())
-            } else {
-                bail!("unknown empty dict type for {}", key)
+                            Ok((format!("Vec<{}>", vec_value), level))
+                        }
+                        "string" => Ok(("Vec<String>".into(), level)),
+                        "boolean" => Ok(("Vec<bool>".into(), level)),
+                        "date" => Ok((format!("Vec<{}>", extract_date_type(value)?), level)),
+                        "number" => Ok((format!("Vec<{}>", extract_number_type(value)?), level)),
+                        "integer" => Ok((format!("Vec<{}>", extract_integer_type(value)?), level)),
+                        "array" => Ok(self.array_recurse_for_type(s, stack, key, level + 1)?),
+                        unknown => {
+                            bail!("unsupported recursive array type \"{unknown}\" for {key}")
+                        }
+                    };
+                }
+                // maybe fallback to serde_json::Value
+                _ => bail!("only support single schema in array {}", key),
             }
+        } else {
+            bail!("missing items in array type")
         }
-        "boolean" => Some("bool".to_string()),
-        "integer" => Some(extract_integer_type(s)?),
-        // think the type we get is the value type
-        x => Some(x.to_upper_camel_case()), // best guess
-    };
-
-    Ok(dict_key)
+    }
 }
 
-// recurse into an array type to find its nested type
-// this recursion is intialised and ended within a single step of the outer recursion
-fn array_recurse_for_type(
-    value: &JSONSchemaProps,
-    stack: &str,
-    key: &str,
-    level: u8,
-) -> Result<(String, u8)> {
+// ----------------------------------------------------------------------------
+// helpers
+fn is_conditions(value: &JSONSchemaProps) -> bool {
     if let Some(items) = &value.items {
-        match items {
-            JSONSchemaPropsOrArray::Schema(s) => {
-                if s.type_.is_none() && s.x_kubernetes_preserve_unknown_fields == Some(true) {
-                    return Ok(("Vec<HashMap<String, serde_json::Value>>".into(), level));
+        if let JSONSchemaPropsOrArray::Schema(props) = &items {
+            if let Some(p) = &props.properties {
+                let type_ = p.get("type");
+                let status = p.get("status");
+                let reason = p.get("reason");
+                let message = p.get("message");
+                let ltt = p.get("lastTransitionTime");
+                let og = p.get("observedGeneration");
+                if type_.is_some()
+                    && status.is_some()
+                    && reason.is_some()
+                    && message.is_some()
+                    && ltt.is_some()
+                    && og.is_some()
```

Comment:
are all of these fields actually required? i mean, the [code upstream](https://github.com/kubernetes/apimachinery/blob/df38a01ea799453419350030b8eb03498842ab76/pkg/apis/meta/v1/types.go#L1523-L1569) says so, but i also know stuff like the `message` and `reason` is not always there, at least on pods (which checks out with [openapi docs](https://docs.rs/k8s-openapi/latest/k8s_openapi/api/core/v1/struct.PodCondition.html)), and the [openapi doc on Condition](https://docs.rs/k8s-openapi/latest/k8s_openapi/apimachinery/pkg/apis/meta/v1/struct.Condition.html) does indicate that at least observed generation is optional.

Reply:
guess it's only `og` that we should skip this check for

Reply:
yeah, we should at the very least skip the check for `og`

Reply:
did https://github.com/kube-rs/kopium/pull/204/commits/c47fe7ffbb8fd93c20ffa2e0b2c4998ca60eb8aa